### DISCUSSION TITLE
Relation won't work if table is named "file"

### DISCUSSION THREAD

**Discussion by bakura10** [mikro-orm/mikro-orm#6161] [General]
### Describe the bug

If a table is named "File", then the ManyToMany relations won't work.

### Reproduction

You can use the following entities:

file.ts:

```
import { Entity, PrimaryKey } from "@mikro-orm/postgresql";

@Entity()
export class File {
  @PrimaryKey()
 id: number;
}
```

project.ts:
```
import { Collection, Entity, ManyToMany, PrimaryKey } from '@mikro-orm/postgresql';
import { File } from './file';

@Entity()
export class Project {
  @PrimaryKey()
  id: number;

  @ManyToMany({ entity: () => File, owner: true })
  files = new Collection<File>(this);
}
```

Then, use the entity repository and try to populate the files relationships:

```
projectRepository.findByCursor({}, { populate: ['files'], first: 20, orderBy: { id: 'desc' } });
```

When inspecting the query, the `file` is transformed to `file2`:

```
─ [query] select "p0".*, "f1"."id" as "f1__id" from "project" as "p0" left join "project_files" as "p2" on "p0"."id" = "p2"."project_id" left join "public"."file2" as "f1" on "p2"."file2_id" = "f1"."id" where "p0"."id" in (select "p0"."id" from (select "p0"."id" from "project" as "p0" left join "project_files" as "p2" on "p0"."id" = "p2"."project_id" left join "public"."file2" as "f1" on "p2"."file2_id" = "f1"."id" group by "p0"."id" order by min("p0"."id"::text) desc limit 21) as "p0") order by "p0"."id" desc [took 2 ms]
─ TableNotFoundException2: select "p0".*, "f1"."id" as "f1__id" from "project" as "p0" left join "project_files" as "p2" on "p0"."id" = "p2"."project_id" left join "public"."file2" as "f1" on "p2"."file2_id" = "f1"."id" where "p0"."id" in (select "p0"."id" from (select "p0"."id" from "project" as "p0" left join "project_files" as "p2" on "p0"."id" = "p2"."project_id" left join "public"."file2" as "f1" on "p2"."file2_id" = "f1"."id" group by "p0"."id" order by min("p0"."id"::text) desc limit 21) as "p0") order by "p0"."id" desc - relation "public.file2" does not exist
```

I suspect that file might be a reserved keyword and might not be used, but the error message is a bit unclear.

Thanks.

### What driver are you using?

@mikro-orm/postgresql

### MikroORM version

6.3.13

### Node.js version

Node.js 20

### Validations

- [X] Read the [Contributing Guidelines](https://github.com/mikro-orm/mikro-orm/blob/master/CONTRIBUTING.md).
- [X] Read the [docs](https://mikro-orm.io/docs).
- [X] Check that there isn't [already an issue](https://github.com/mikro-orm/mikro-orm/issues) that reports the same bug to avoid creating a duplicate.
- [X] Check that this is a concrete bug. For Q&A open a [GitHub Discussion](https://github.com/mikro-orm/mikro-orm/discussions) or join our [Discord](https://discord.gg/w8bjxFHS7X).
- [X] The provided reproduction is a [minimal reproducible example](https://stackoverflow.com/help/minimal-reproducible-example) of the bug.

**Comment by lsh4711** [comment]
https://github.com/lsh4711/reproduction/tree/6155
Would you like to try this?
The issue you mentioned isn't occurring.
The query that is being generated is as follows.
```
select "p0".*, "f1"."id" as "f1__id" from "project" as "p0" left join "project_files" as "p2" on "p0"."id" = "p2"."project_id" left join "public"."file" as "f1" on "p2"."file_id" = "f1"."id" where "p0"."id" in (select "p0"."id" from (select "p0"."id" from "project" as "p0" left join "project_files" as "p2" on "p0"."id" = "p2"."project_id" left join "public"."file" as "f1" on "p2"."file_id" = "f1"."id" group by "p0"."id" order by min("p0"."id") desc limit 21) as "p0") order by "p0"."id" desc
```
You can provide a minimal reproduction of the issue by forking [reproduction](https://github.com/mikro-orm/reproduction).

**Comment by B4nan** [comment]
Sounds like you are using some bundler, and your class ends up called `File2` without you realizing. The ORM won't rename your entity like this.

**Reply by bakura10** [reply]
I'm using ESBuild here. The strange thing is that this is reproducible even if I explicitly pass a table name:

```
@Entity({ tableName: 'file' })
export class File {
}
```

Wouldn't MikroORM use this info, independently of the class name?

**Reply by B4nan** [reply]
I wonder how are you using esbuild with decorators, I thought they don't support the legacy spec (and we don't really support the ESM spec yet).

**Comment by bakura10** [comment]
Thanks for the hint @B4nan 

For anyone facing this issue, this is caused to ESBuild renaming symbols. To avoid this problem, you must use the --keep-names flag for esbuild.
